deep-learning/src/main/python/synapse/ml/dl/DeepVisionModel.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in project root for information.
+
+import numpy as np
+import torch
+import torchvision.transforms as transforms
+from horovod.spark.lightning import TorchModel
+from PIL import Image
+from pyspark.ml.base import _PredictorParams
+from pyspark.ml.param.shared import Param, Params
+from pyspark.sql.functions import col, udf
+from pyspark.sql.types import DoubleType
+from synapse.ml.dl.utils import keywords_catch
+
+
+class DeepVisionModel(TorchModel, _PredictorParams):
+
+    transform_fn = Param(
+        Params._dummy(),
+        "transform_fn",
+        "A composition of transforms used to transform and augnment the input image, should be of type torchvision.transforms.Compose",
+    )
+
+    @keywords_catch
+    def __init__(
+        self,
+        history=None,
+        model=None,
+        input_shapes=None,
+        optimizer=None,
+        run_id=None,
+        _metadata=None,
+        loss=None,
+        loss_constructors=None,
+        # diff from horovod
+        transform_fn=None,
+        labelCol="label",
+        featuresCol="features",
+        predictionCol="prediction",
```

Comment:
I think we should either have all the params be settable with camel or none. Having half and half seems like it will confuse folks

Reply:
I'll create params instead of extending _PredictorParams.